gen/flights/flightplan.py

```diff
@@ -1703,36 +1852,78 @@ def generate_cas(self, flight: Flight) -> CasFlightPlan:
         )
 
     def generate_refueling_racetrack(self, flight: Flight) -> RefuelingFlightPlan:
-        location = self.package.target
+        is_deployment_support: bool = False
 
-        closest_boundary = self.threat_zones.closest_boundary(location.position)
-        heading_to_threat_boundary = Heading.from_degrees(
-            location.position.heading_between_point(closest_boundary)
-        )
-        distance_to_threat = meters(
-            location.position.distance_to_point(closest_boundary)
-        )
-        orbit_heading = heading_to_threat_boundary
+        package_waypoints = self.package.waypoints
+        assert package_waypoints is not None
 
-        # Station 70nm outside the threat zone.
-        threat_buffer = nautical_miles(70)
-        if self.threat_zones.threatened(location.position):
-            orbit_distance = distance_to_threat + threat_buffer
+        racetrack_half_distance = Distance.from_nautical_miles(20).meters
+
+        patrol_duration = timedelta(hours=1)
+
+        # Determine if tanker is supporting a package.
+        if package_waypoints is not None and package_waypoints.refuel is not None:
+            is_deployment_support = True
+
+        # Tanker is supporting a package
+        if is_deployment_support:
+
+            assert package_waypoints.refuel is not None
+
+            # TODO: Only consider aircraft that can refuel with this tanker type.
+            refuel_time_minutes = 5
+            for flight in self.package.flights:
+                flight_size = flight.roster.max_size
+                refuel_time_minutes = refuel_time_minutes + 4 * flight_size + 1
```

Comment:
If tankers are planned when it is thought that the package doesn't have the fuel to make it home, then there does need to be enough time for the aircraft to refuel.  I do feel like the time formula is excessive but the DCS AI has been unpredictable.

Reply:
I suspect it's also the case that in a large enough package where enough aircraft do need to refuel, you either have sent multiple tankers that they're hopefully smart enough to shard across, or the _tanker_ is going to run out of gas before 40 minutes elapses.

My idea of a time limit wouldn't work anyway since the aircraft at the back of the line wouldn't continue on when they see the wait, they'd wait 30 minutes and _then_ go home. What I was thinking is that they'd ideally see that they won't be able to get gas for 40 minutes but could bingo home in 30. I think you're right though and the AI isn't predictable enough for us to try anything better here.